### DISCUSSION TITLE
How to transform velocity.h5 to geographic coordinates and convert the velocity in geographic coordinates to a simple xyz.txt (longitude latitude velocity)

### DISCUSSION THREAD

**Discussion by DD-ZHAO** [insarlab/MintPy#1296] [Q&A]
Hello everyone, when I use geocode.py to convert velocity. h5, it prompts 'no lookup table found'. How can I convert velocity to the geographic coordinate ? Furthermore, how can I convert the velocity in the geographic coordinate to xyz.txt (longitude latitude velocity)? Thank you for your time!

**Comment by fgdiazm** [comment]
Hello,
Try to generate a geotiff with

save_gdal.py geo_velocity.h5

With the .Tiff file you could project it in QGIS.

**Reply by DD-ZHAO** [reply]
Hello, thank you for your reply. After mintpy's normal end, I got velocity. h5, but not geo_velocity. h5. How can I get geo_velocity. h5 in geographic coordinate.

**Reply by keepfighting666** [reply]
geocode step

**Reply by ditafaith** [reply]
Hi @DD-ZHAO, have you solve this issue?
It seems similar with mine

**Comment by AlekseyDmitriev** [comment]
To convert the velocity.h5 file into geographic coordinates, I use one of the source files as a reference. From it, I extract the geographic information and apply it to the datasets in the velocity.h5 file. The enclosed script will help you with this task.
[split_hdf5_to_files.py](https://github.com/user-attachments/files/24376559/split_hdf5_to_files.py)

**Reply by ditafaith** [reply]
Hi @AlekseyDmitriev Thank you for sharing the solution. I am not understand what do you mean " I use one of the source files as a reference. From it, I extract the geographic information". Sorry for my bad but better to give an example.

**Reply by AlekseyDmitriev** [reply]
Sorry for such a brief answer. Let me try to explain in more detail.
I use data from ASF HyP3 for processing with MintPy, i.e., unwrapped phases, coherence, etc. For correct processing, they must be the same size, meaning they are cropped to cover the common area for all. These prepared files for processing can be used as reference files — any of them, such as *_unw_phase.tif or *_corr.tif. Only geographic information is extracted from the file. The attached script provides an example of how to do this. The REF_FILE variable in this case points to one of the unwrapped phase files that was used to calculate velocity.h5.

**Reply by ditafaith** [reply]
Thank you, I got your points.
Then, user need have to change the following path,
DATA_PATH = Path('path/to/data_dir')
    REF_FILE = "path/to/reference_file/S1AA_20170502T194057_20170514T194058_VVP012_INT80_G_weF_ED9F_unw_phase_clip.tif"

**Reply by AlekseyDmitriev** [reply]
You're absolutely right.
Yes, the user needs to change the following paths in the script:
DATA_PATH: This should point to the directory containing the velocity.h5 file.
REF_FILE: This must be the full path to one of the prepared and cropped input GeoTIFF files that were used to generate that specific velocity.h5 file. For example:
.../your_project/S1AA_20170502_..._unw_phase.tif or
.../your_project/S1AA_20170502_..._corr.tif

**Reply by ditafaith** [reply]
Thank you very much @AlekseyDmitriev. This code successfully generated geotif files in the velocity directory.
I think this code could be modified slightly to convert all cropped images *.tif files obtained from hyp3. Then, we may proceed these inputs in MintPy using geotif format.
